components/inputs/input-date.js

```diff
@@ -212,12 +222,35 @@ class InputDate extends LocalizeStaticMixin(LitElement) {
 		super.updated(changedProperties);
 
 		changedProperties.forEach((oldVal, prop) => {
-			if (prop === 'value') {
-				this._formattedValue = this.value ? formatISODateInUserCalDescriptor(this.value) : '';
+			if (prop === '_dateTimeDescriptor' || prop === 'value') {
+				this._setFormattedValue();
 			}
 		});
 	}
 
+	_getContentWidth() {
+		const text = document.createElement('div');
+		document.body.appendChild(text);
+		text.style.fontSize = '0.8rem';
+		text.style.width = 'auto';
+		text.style.position = 'absolute';
+
+		// in some languages (e.g., fr) placeholderWidth is bigger, in others (e.g., zh) contentWidth is bigger
+		text.innerHTML = (this._dateTimeDescriptor.formats.dateFormats.short).toUpperCase();
+		const placeholderWidth = text.clientWidth;
+		text.innerHTML = formatISODateInUserCalDescriptor('2020-12-20');
+		const contentWidth = text.clientWidth;
+
+		let emptyStateWidth = 0;
+		if (this.emptyStateText) {
+			text.innerHTML = this.emptyStateText;
+			emptyStateWidth = text.clientWidth;
+		}
+		const width = Math.max(placeholderWidth, contentWidth, emptyStateWidth);
+		this._contentWidth = width + 10;
```

Comment:
What's the `+ 10` for? If you're seeing a mismatch, it could be because our inputs have `letter-spacing: 0.2rem`...

Reply:
Without it we get the `...` at the end instead of the final letters/numbers. Is there a better alternative?

Reply:
I'd just like to know the root cause and why `10` works but `8` or `9` doesn't. If the user has a larger font size, do you need a larger value here?

Reply:
So it turned out to be a combination of needing to use `letter-spacing`, adding border width, and needing to do some rounding up. The mysterious value has been eliminated🎉